File: web/features/payments/domain/entities/payment-concept.ts
export interface PaymentConcept {
  id: string;
  school_id: string;
  name: string;
  amount: number;
  is_recurring: boolean;
  state: boolean;
}

export interface PaymentConceptCreate {
  name: string;
  amount: number;
  is_recurring?: boolean;
}

export interface PaymentConceptUpdate {
  name?: string;
  amount?: number;
  is_recurring?: boolean;
  state?: boolean;
}

File: server/app/modules/payments/models/payment_concept.py
from datetime import time
from typing import Optional
from uuid import UUID

from sqlalchemy import Index, text
from sqlmodel import Field

from app.core.base_model import UUIDBaseModel


# Modelo de concepto de pago (ej. cuota)
class PaymentConcept(UUIDBaseModel, table=True):
    __tablename__ = "payment_concepts"
    __table_args__ = (
        Index(
            "uq_payment_concepts_school_name_active",
            "school_id",
            "name",
            unique=True,
            sqlite_where=text("state = 1"),
            postgresql_where=text("state = true"),
        ),
    )

    school_id: UUID = Field(foreign_key="school.id", index=True)
    name: str = Field(index=True, max_length=100)
    amount: float = Field(default=0.0)
    is_recurring: bool = Field(default=True)
